fix(js): Include tokens for arrow function params

nmote opened the pull request:
Previously these were left out, so the location of `$X` in the tests added would be wrong, leading to incorrect autofix results.

Helps #7353

Test plan: Automated tests

PR checklist:

- [x] Purpose of the code is [evident to future readers](https://semgrep.dev/docs/contributing/contributing-code/#explaining-code)
- [x] Tests included or PR comment includes a reproducible test plan
- [x] Documentation is up-to-date
- [x] A changelog entry was [added to changelog.d](https://semgrep.dev/docs/contributing/contributing-code/#adding-a-changelog-entry) for any user-facing change
- [x] Change has no security implications (otherwise, ping security team)

If you're unsure about any of this, please see:

- [Contribution guidelines](https://semgrep.dev/docs/contributing/contributing-code)!
- [One of the more specific guides located here](https://semgrep.dev/docs/contributing/contributing/)

brandonspark commented on languages/javascript/ast/ast_js.ml:
```diff
@@ -429,7 +429,7 @@ and function_definition = {
   (* less: move that in entity? but some anon func have attributes too *)
   f_attrs : attribute list;
-  f_params : parameter list; (* TODO: bracket *)
+  f_params : parameter list bracket; (* TODO: bracket *)
```
could you remove this TODO?

brandonspark reviewed (approved):
looks good!

nmote committed:
fix(js): Include tokens for arrow function params

Previously these were left out, so the location of `$X` in the tests
added would be wrong, leading to incorrect autofix results.

Helps #7353

Test plan: Automated tests
